src/core_plugins/kibana/public/dashboard/panel/panel.js

```diff
@@ -117,6 +118,20 @@ uiModules
           $scope.$watchCollection('panel.sort', function () {
             $scope.saveState();
           });
+
+          $scope.addColumn = function addColumn(columnName) {
+            $scope.savedObj.searchSource.get('index').popularizeField(columnName, 1);
+            columnActions.addColumn($scope.panel.columns, columnName);
+          };
+
+          $scope.removeColumn = function removeColumn(columnName) {
+            $scope.savedObj.searchSource.get('index').popularizeField(columnName, 1);
```

Comment:
Interesting. Is this new? A quick test seems to indicate that previously adjusting the columns via dashboard didn't affect popularity score.  Makes sense to add it here, though.